pkg/apis/events/v1alpha1/cloudstoragesource_validation.go

```diff
@@ -58,6 +59,13 @@ func (current *CloudStorageSourceSpec) Validate(ctx context.Context) *apis.Field
 		}
 	}
 
+	if current.ServiceAccount != nil {
```

Comment:
you should also validate that you cannot have serviceAccount and secret set at the same time.